walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -183,4 +184,33 @@ public static TimeRange fromStartEnd(int start, int end, boolean inclusive) {
   public static TimeRange fromStartDuration(int start, int duration) {
     return new TimeRange(start, duration);
   }
+
+  /**
+   * Uses a branchless binary search to find the TimeRange with the biggest starting point,
+   * that is <= to the starting point of target.
+   * It searches the index of the target bit by bit, starting with the MSB and going
+   * down to the LSB of the index.
+   * @param list the list of TimeRanges to search from.
+   * @param target the TimeRange to be searched
+   * @return if either the list or the target are null, returns -1. If the list
```

Comment:
The first two cases of the return value (when list or target are null -> -1; when the list is empty -> 0) are a bit awkward: looks like the calling code should not run into those (there is no check for -1 or 0 in the calling code) and this choice is a bit arbitrary.
If that's the case (calling code shouldn't run into those cases), I'd suggest this to throw an IllegalArgumentException with a message reflecting the issue in those cases. That makes the interface a bit more clear by eliminating the special return values for these edge cases and providing a guarantee in the calling code that the index returned (if there was no exception thrown) can be used to safely index into `list`.